src/cargo/core/workspace.rs

```diff
@@ -103,6 +103,12 @@ pub struct Workspace<'gctx> {
     /// file. This is set for `cargo install` without `--locked`.
     ignore_lock: bool,
 
+    /// If `true`, then the resolver will not update the `Cargo.lock` file and
+    /// return an error if the lockfile is missing or out of date, similar
+    /// to the `GlobalContext::locked()` behaviour. Note that
```

Comment:
There are already too many fields controlling the `locked` behavior. Per Ed's suggestion, I'd like to see if we can add one line explaining this field is independent from `--locked` CLI flag. Perhaps something like:

```suggestion
    /// If `true`, then the resolver will not update the `Cargo.lock` file and
    /// return an error if the lockfile is missing or out of date.
    ///
    /// This field is independent from any flags like `--locked` or `--frozen`.
    /// Use [`GlobalContext::locked()`] whenever you have an access to mutable
    /// to [`GlobalContext`], such as at Command arg parsing level.
```